client/src/main/java/org/apache/falcon/resource/SchedulableEntityInstance.java

```diff
@@ -170,6 +170,11 @@ public int compareTo(SchedulableEntityInstance o) {
             return result;
         }
 
+        result = this.tags.compareTo(o.tags);
+        if (result != 0) {
+            return result;
+        }
+
```

Comment:
This could potentially throw NPE since tags could be null. Instead, you can use null-safe comparison provided by ObjectUtils class -
result = ObjectUtils.compare(this.tags, o.tags);